ppx/ppx_js/lib_internal/ppx_js_internal.ml

```diff
@@ -585,7 +585,14 @@ let preprocess_literal_object mappper fields :
         let body = format_meth (mappper body) in
         let rec create_meth_ty exp =
           match exp.pexp_desc with
-          | Pexp_fun (label, _, _, body) -> Arg.make ~label () :: create_meth_ty body
+          | Pexp_function (params, _, Pfunction_body body) ->
+              let params =
+                List.filter_map params ~f:(function
+                  | { pparam_desc = Pparam_newtype _; _ } -> None
+                  | { pparam_desc = Pparam_val (label, _, _arg); _ } ->
+                      Some (Arg.make ~label ()))
+              in
+              params @ create_meth_ty body
```

Comment:
I'm not sure we need to recurse here.

Reply:
@hhugo the reason why I added recursion was for cases like this:

```ocaml
class x =
  object
    method foo a = fun b -> a + b
  end
```

due to https://github.com/ocaml/ocaml/pull/12236 preserving both `Pexp_function` nodes in the AST. I'm not very familiar with how JSOO handles this case, so I trust you to know whether it's unnecessary.